### ISSUE TITLE
intermittent build failures due to submodule

### ISSUE THREAD

**Issue by smaiyaki** [Shippable/support#1214]
my configuration is as thus:

* In the repository, I have added a submodule. I have however not staged the submodule to avoid my repo from being unnecessary big
* In shippable, in the before_script tag, I added these 2 steps:
  * git submodule init
  * git sumodule update
* At this stage, the submodule should be checkout during the build
* I will then cd in to the submodule and run a script that starts another build in the background

While this step works in most of the cases, it occasionally fails to run my script with the error that the script file could not be found. I have added some extra line to list out all the file listing before running the script file and it shows that the submodule has not been cloned and thus the failure. a rebuild or a couple of rebuilds will fix the issue

In summary, the submodule feature is a buggy I guess

**Comment by manishas** [comment]
@smaiyaki we're writing up a post on submodules this week and I'll point you to it as soon as it's available.

**Comment by Nalum** [comment]
Any update to when this blog post is coming out? I'm having trouble with submodule initialization. I've added the deploy key used on the main repo to the submodule but it's still not able to get the submodule. I'm using bitbucket in case that has any say in the matter.

**Comment by manishas** [comment]
this is an old issue and we've revamped our build system since then.
